fmanimashaun/school_library
# OOP school library: preserve data

# Implement Data Persistence with JSON

## Description

This PR addresses the requirement of persisting application data using JSON files. It ensures that all data (people, books, and rentals) are saved into separate JSON files upon application exit, and are loaded back into the application upon startup.

## Changes Made

1. Added `json` library to the project for handling JSON data.
2. Implemented `to_hash` method in `Book`, `Person`, and `Rental` classes to convert objects to hash format.
3. Created a `Store` module with methods to handle saving to file and retrieving from file.
4. Integrated the `Store` module methods into the main application flow to ensure data is saved on exit and loaded on startup.

## How to Test

1. Run the application and perform various operations like adding books, people, and rentals.
2. Exit the application.
3. Check the project directory for `people.json`, `books.json`, and `rentals.json` files.
4. Start the application again.
5. All previously added data should be loaded and available for use in the application.
